### ISSUE TITLE
`check-template-names` reports on @template names that are not a single capital letter

### ISSUE THREAD

**Issue by RobertAKARobin** [gajus/eslint-plugin-jsdoc#1352]
## ESLint sample

```js
/**
 * @template Input
 * @param {Input} input
 * @returns {Input}
 */
function echo(input) {
	return input;
}

const myString = echo(`hello`);
const myNum = echo(32);
```

## Expected behavior

Expected no errors.

## Actual behavior

`check-template-names` reports:

`@template Input is not in use`

## ESLint Config

```js
import jsdoc from 'eslint-plugin-jsdoc';

export default [
	{
		files: [`**/*.js`],
		plugins: {
			jsdoc,
		},
		rules: {
			'jsdoc/check-template-names': `error`,
		},
	},
];
```

## Environment

- Node version: 23.6.0
- ESLint version 9.19.0
- `eslint-plugin-jsdoc` version: 50.6.2

**Comment by RobertAKARobin** [comment]
I believe the culprit is this RegEx, `/^[A-Z]$/`:

https://github.com/gajus/eslint-plugin-jsdoc/blob/main/src/rules/checkTemplateNames.js#L40

For some reason many people name their type variables with a single capital letter, which would match this RegEx. However I find such variables very hard to read and so prefer to use more descriptive type names.

Oddly, it looks like descriptive template names _are_ allowed but only in Typescript files:

https://github.com/gajus/eslint-plugin-jsdoc/blob/main/test/rules/assertions/checkTemplateNames.js#L538

**Comment by brettz9** [comment]
This should now be fixed.

Note that when using templates within JSDoc types, the `require-template` rule still only checks for names that are a single letter. The reason for this is that we otherwise have no idea in that rule which name is a template and which is some other variable (we just hope the user doesn't use single letters for normal variables). But indeed, we don't need to restrict things in this rule.

**Comment by RobertAKARobin** [comment]
Holy cow, that was fast! Thanks!
